types/angular/jqlite.d.ts

```diff
@@ -651,6 +651,7 @@ interface JQuery {
 
     // Undocumented
     length: number;
+    [index: number]: TElement;
```

Comment:
That sounds like a reasonable workaround.

I'm still not sure I understand how your build passed to begin with.